canonical/react-components
# feat: adjust contextual dropdown vertical position

## Done
This adjusts contextual dropdown vertical position based on the available space in the parent container. It's assuming that the container that should be used for the measurement is the first first scrollable parent of the dropdown. 

This is fairly computationally expensive, but because it happens on open it shouldn't cause issues in real-world scenarios.

- feat: adjust contextual dropdown vertical position
- add missing Button import to ContextualMenu story 

## QA
- Go to http://localhost:9009/?path=/story/contextualmenu--overflowing-container
- Verify that the first dropdown displays its content below the button and second one displays the content above.

### Storybook

To see rendered examples of all react-components, run:

```shell
yarn start
```

### QA in your project

from `react-components` run:

```shell
yarn build
npm pack
```

Install the resulting tarball in your project with:

```shell
yarn add <path-to-tarball>
```

### QA steps

- Steps for QA.

## Fixes

Fixes: https://warthogs.atlassian.net/browse/MAASENG-2428
